orchestrator/graphql/schema.py

```diff
@@ -130,52 +135,64 @@ def custom_context_dependency(
 
 
 def get_context(
+    graphql_models: StrawberryModelType,
     broadcast_thread: ProcessDataBroadcastThread | None = None,
 ) -> Callable[[OrchestratorContext], Coroutine[Any, Any, OrchestratorContext]]:
     async def _get_context(
         custom_context: OrchestratorContext = Depends(custom_context_dependency),  # noqa: B008
     ) -> OrchestratorContext:
         custom_context.broadcast_thread = broadcast_thread
+        custom_context.graphql_models = graphql_models
         return custom_context
 
     return _get_context
 
 
-GRAPHQL_MODELS_INITIAL = dict(GRAPHQL_MODELS)
+INTERNAL_GRAPHQL_MODELS: StrawberryModelType = {}
+
+
+def internal_graphql_model_test_logic(graphql_models: dict) -> dict:
+    """Necessary Graphql logic for unit tests since we create and delete subscription models in each test and graphql expect static models.
+
+    We create and destroy the subscription models in each test. (which makes the pydantic model technically not the same after its re-created)
+    Strawberry will error saying that a nested product block can't be resolver and if you are actually using strawberry.field, This should never happen outside of testing.
+
+    returns INTERNAL_GRAPHQL_MOEDLS if environment is 'TESTING' and graphql_models if isn't
+    """
+    if app_settings.ENVIRONMENT == "TESTING":
+        INTERNAL_GRAPHQL_MODELS.update(graphql_models)
+        return INTERNAL_GRAPHQL_MODELS
+    return graphql_models
 
 
 def create_graphql_router(
     query: Any = Query,
     mutation: Any = Mutation,
     register_models: bool = True,
```

Comment:
This was already there, but please don't add booleans to function defs. Just create another function, and have them call some shared code.

Reply:
will keep it in mind